### DISCUSSION TITLE
MPI_Abort(MPI_COMM_WORLD, 1) - process 0

### DISCUSSION THREAD

**Discussion by ritamcj** [idaholab/moose#20581] [Q&A General] [answered]
Hello everyone,
I am simulating uniaxial compression for a dual phase material using tensor mechanics module. After some amount of deformation, I see an `MPI_Abort error`. I compiled my files in debug mode and ran it in GDB. The output is attached in the pdf [debug_17032022_17:04.pdf](https://github.com/idaholab/moose/files/8284363/debug_17032022_17.04.pdf). Could someone kindly explain "**You requested to reuse the nonlinear residual vector as the base vector for computing the action of the matrix-free Jacobian, but the vectors are not the same. Your physics must have states; either remove the states from your code or make sure that you set_mf_reuse_base(false)**". Does "state" mean state variables? Also, I couldn't find where to put `set_mf_reuse_base = false` in my input file. My executioner block is as follows:
```
[Preconditioning]
  [./SMP]
    type = SMP
    full = true
  [../]
[]

[Executioner]

  [./TimeStepper]
    type = IterationAdaptiveDT
    dt = 0.01
    growth_factor = 1.0000001
    cutback_factor = 0.9
    optimal_iterations = 15
    iteration_window = 2
    reject_large_step = true
    reject_large_step_threshold = 0.1
  [../]


  type = Transient
  nl_max_its = 15
  scheme = bdf2
  solve_type = PJFNK
  petsc_options = -snes_ksp_ew
  petsc_options_iname = '-pc_type'
  petsc_options_value = 'lu'
  l_max_its = 40
  l_tol = 1.0e-3
  nl_rel_tol = 1.0e-5
  start_time = 0.0
  num_steps = 200000
  nl_abs_tol = 1e-5
  line_search = none
  automatic_scaling = true
  scaling_group_variables = 'disp_x disp_y disp_z'
  [./Predictor]
    type = SimplePredictor
    scale = 1.0
    skip_after_failed_timestep = true
  [../]
[]
```

Apart from this, I don't see "Bad file descriptor" error when I run my file in opt mode. I look forward to your advice.
Thanks and regards,
Ritam

**Comment by GiudGiud** [comment, marked as answer]
Hello

This is a parameter in the Executioner, see the solver parameters
https://mooseframework.inl.gov/source/executioners/Transient.html

This error means `PETSc saw the nonlinear function change from nonlinear and linear iterations. That is not common, and in moose, that occurs primarily in mechanics contact module because of contact` (@fdkong) 
Do you have contact in your simulation? 

Guillaume

**Reply by ritamcj** [reply]
Hi Guillaume,
I do not have contact in my simulation. However, I am applying uniaxial compression along z-axis on a dual phase grain structure with phase-2 situated at phase-1 grain boundaries. The initial and final structures (when the simulation fails) are attached. If the phases are coming in contact after some amount of deformation, it makes sense to me that the simulation fails. I will try snesmf_reuse_base = false and check. I'll also simulate the same problem for uniaxial tension and see whether I see the same error. Many thanks for your suggestions.
Ritam
![initial_grain_structure](https://user-images.githubusercontent.com/81842276/158876999-b5072179-7cd7-41e3-a94c-5d2ed631a627.png)

 
![final_grain_structure](https://user-images.githubusercontent.com/81842276/158879076-93d1ca92-6cbe-4ea1-a555-cd8b23786a7d.png)

**Reply by ritamcj** [reply]
Hi everyone,
Just wanted to inform that MPI Abort error was due to improper node set generation inside Moose using `BoundingBoxNodeSetGenerator`. There were inaccuracies in the coordinates of the bottom left and top right corners of faces. I modified the mesh generation method and am now reading the mesh from an inp file from Dream3D. @GiudGiud Thank you for your help.
Warm regards,
Ritam
